packages/compass-editor/src/codemirror/validation-autocompleter.ts

```diff
@@ -0,0 +1,176 @@
+import type { CompletionSource } from '@codemirror/autocomplete';
+import { snippetCompletion } from '@codemirror/autocomplete';
+import { completer } from '../autocompleter';
+import type { CompletionResult } from '../autocompleter';
+
+import {
+  resolveTokenAtCursor,
+  getAncestryOfToken,
+  ARRAY_ITEM_REGEX,
+} from './utils';
+
+const rootJsonSchemaCompletion = (from: number, to: number) => ({
+  filter: false,
+  from,
+  to,
+  options: [
+    snippetCompletion('{\n\t$jsonSchema: {\n\t\t${}\n\t}\n}', {
+      label: '$jsonSchema',
+      info: 'json-schema validation object',
+    }),
+  ],
+});
+
+const createCompletions = (
+  completions: CompletionResult[],
+  text: string,
+  from: number,
+  detail?: string
+) => {
+  return {
+    from,
+    filter: false,
+    options: completions
+      .filter((completion) => {
+        return completion.value.toLowerCase().startsWith(text.toLowerCase());
+      })
```

Comment:
Unless you are doing this for some specific reason, a recommended approach from codemirror docs is to just return all results with `validFor` instead of doing filtering yourself, [here's an example](https://github.com/mongodb-js/compass/pull/4163/files#diff-09b51c2c02afdec3ad7c224bb84ec90e1878d0724ca903f904c9fa84cfdde309R47-R64)